### ISSUE TITLE
Support maven deploy style submission

### ISSUE THREAD

**Issue by mbax** [SpongePowered/Ore#239]
Proposal: Support, in Ore, for uploading files (and signature) in the style one would do with the Maven deploy plugin.

I believe the plugin supports release and snapshot channels, so in order to enable this feature on a project someone would have to designate which channel is which. Maybe even default to release supported on the default channel.

**Comment by Grinch** [comment]
In talks between @Zidane and @mbax. We've also agreed on the following.

 * Any channel created should be able to toggle whether or not submissions to said channel will be reviewed. If a user toggles this from non-reviewed to reviewed then only submissions passed that will be reviewed. Warning pages will still apply as needed based on approval status.
 * All uploads should be signed. No exceptions.

**Comment by mbax** [comment]
Just for connecting things, reviewed/non-reviewed is #228

**Comment by windy1** [comment]
Yes this has been my plan from pretty much day 1 however I would like to get the current platform really solid before this happens.
